sdv/timeseries/deepecho.py

```diff
@@ -67,7 +67,8 @@ def _fit(self, timeseries_data):
 
         data_types = list()
         context_types = list()
-        for field, meta in self._metadata.get_fields().items():
+        for field in self._entity_columns + self._data_columns:
+            meta = self._metadata.get_fields()[field]
```

Comment:
I would capture the `fields_metadata` in a variable before the loop to avoid having to call `self._metadata.get_fields()` at each iteration.